pandas/tests/frame/test_constructors.py

```diff
@@ -717,6 +717,16 @@ def test_constructor_period_dict(self):
         assert df["a"].dtype == a.dtype
         assert df["b"].dtype == b.dtype
 
+        # list of periods
+        df = DataFrame(
+            {
+                "a": pd.Period("2012-01", freq="M"),
+                "b": pd.Period("2012-02-01", freq="D"),
+            }
+        )
+        assert df["a"].dtype == "period[M]"
```

Comment:
construct the expected frame and use assert_frame_equal

Reply:
Since this is testing the construction of dfs, it would seem counterintuitive and potentially problematic to test via a different construction method.

For example, if I were to assert that
```python
df = DataFrame(
            {
                "a": pd.Period("2012-01", freq="M"),
                "b": pd.Period("2012-02-01", freq="D"),
            }
        )
```
is equivalent to
```python
pd.DataFrame(
    [[pd.Period("2012-01", freq="M"), pd.Period("2012-02-01", freq="D")]],
    columns=['a', 'b']
)
```
Then if there was an issue that effected both construction via a dict and construction via an array, the test would not pick up on it.

Reply:
@jreback thoughts?